builtin/providers/azure/resource_azure_hosted_service.go

```diff
@@ -0,0 +1,190 @@
+				Required:    true,
+				ForceNew:    true,
+				Description: parameterDescriptions["location"],
+			},
+			"url": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ephemeral_contents": &schema.Schema{
+				Type:     schema.TypeBool,
+				Required: true,
+				DefaultFunc: func() (interface{}, error) {
+					return false, nil
+				},
+				Description: parameterDescriptions["ephemeral_contents"],
+			},
+			"status": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"reverse_dns_fqdn": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: parameterDescriptions["reverse_dns_fqdn"],
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: parameterDescriptions["description"],
+			},
+			"default_certificate_thumbprint": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: parameterDescriptions["default_certificate_thumbprint"],
+			},
+		},
+	}
+}
+
+// resourceAzureHostedServiceCreate does all the necessary API calls
+// to create a hosted service on Azure.
+func resourceAzureHostedServiceCreate(d *schema.ResourceData, meta interface{}) error {
+	azureClient := meta.(*AzureClient)
+	managementClient := azureClient.managementClient
+	hostedServiceClient := hostedservice.NewClient(managementClient)
+
+	serviceName := d.Get("service_name").(string)
+	location := d.Get("location").(string)
+	reverseDNS := d.Get("reverse_dns_fqdn").(string)
+	description := d.Get("description").(string)
+
+	// set the label as the resource's ID:
+	label := getRandomStringLabel(20)
+	d.SetId(label)
+
+	err := hostedServiceClient.CreateHostedService(
+		hostedservice.CreateHostedServiceParameters{
+			ServiceName:    serviceName,
+			Location:       location,
+			Label:          label,
+			Description:    description,
+			ReverseDNSFqdn: reverseDNS,
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("Failed defining new Azure hosted service: %s", err)
+	}
+
+	return nil
+}
+
+// resourceAzureHostedServiceExists checks whether a hosted service with the
+// given service_name already exists on Azure.
+func resourceAzureHostedServiceExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	azureClient := meta.(*AzureClient)
+	hostedServiceClient := hostedservice.NewClient(azureClient.managementClient)
+
+	log.Println("[INFO] Querying for hosted service existence.")
+	serviceName := d.Get("service_name").(string)
+	_, err := hostedServiceClient.GetHostedService(serviceName)
+	if err != nil {
+		if management.IsResourceNotFoundError(err) {
+			// it means that the hosted service has been deleted in the meantime...
+			d.SetId("")
+			return false, nil
+		} else {
+			return false, fmt.Errorf("Failed to query for hosted service name availability: %s", err)
+		}
+	}
+
+	return true, nil
+}
```

Comment:
Or alternatively, you could leave `Exists` and remove the short circuit from `Read` - that's another valid use case of `Exists` - to allow `Read